core/rawdb/accessors_indexes.go

```diff
@@ -27,28 +29,36 @@ import (
 
 // ReadTxLookupEntry retrieves the positional metadata associated with a transaction
 // hash to allow retrieving the transaction or receipt by hash.
-func ReadTxLookupEntry(db ethdb.Reader, hash common.Hash) common.Hash {
+func ReadTxLookupEntry(db ethdb.Reader, hash common.Hash) *uint64 {
```

Comment:
Would it be worthwhile to have two methods? One that returns a `uint64` and one that returns `hash`? That way, we would sometimes not needs to do lookup tx -> hash, read hashto number -> read cannon hash (number). 

It would also make the code for the callers a bit simpler

Reply:
@holiman Could you point me to the code that you think would benefit from this? How performance critical is that code as well?

Reply:
I was thinking about `ReadTransaction` and `ReadReceipt`. 
They do
`ReadTxLookupEntry, ReadCanonicalHash,ReadBody `. 
If it's an old tx, it would internally be 
`read a hash, ReadHeaderNumber, ReadCanonicalHash,ReadBody `. 
which could be 
`read a hash -> ReadBody `. 
Maybe it would just overcomplicate things

Reply:
I agree that legacy databases pay a penalty here. If a user was using the client for constantly looking up transactions they may feel the effects of this slowdown. However, if this was a serious user I would probably expect them to resync. We could also offer a CLI tool to do these various database upgrades. @karalabe let me know if this is something that would be valuable to have.